lusid-sdk-python/tests/business_agility_test_environment.py

```diff
@@ -0,0 +1,662 @@
+        then populate our initial holdings on these instruments.
+        We will test that we have correctly created the instruments and added the holdings to our portfolio
+        """
+
+        '''
+        Rather than using separate update and insert methods, LUSID uses an upsert method for may of its operations.
+
+        This inserts a record if it does not already exist and modifies an existing record if it does already exist.
+
+        We can upsert our instruments individually or in a batch. Either way the request should be in the form of a
+        dictionary with an arbitrary reference name for the instrument as the key and an InstrumentDefinition object
+        as the value.
+
+        The InstrumentDefinition requires a name for the instrument which is a string. In this case we will use the 
+        same name as the arbitrary reference key.
+
+        It also requires a dictionary containing at least one unique identifier for the asset. The key for this 
+        dictionary is the name of the identifier, in this case we will use 'ClientInternal'. The value is a string which
+        is the identifier. The allowed identifiers currently supported are:
+
+            - ClientInternal
+            - Figi
+        '''
+        # Initialise our batch upsert request
+        batch_upsert_request = {}
+        # Using then initial holdings information, create our batch
+        for instrument_name in self.transferred_instruments:
+            batch_upsert_request[instrument_name] = models.InstrumentDefinition(name=instrument_name,
+                                                                                identifiers=
+                                                                                   self.transferred_instruments[
+                                                                                       instrument_name]['identifiers'])
+        # Upsert our batch
+        batch_upsert_response = self.client.upsert_instruments(requests=batch_upsert_request)
+
+        # Tests - Confirm that the response is as expected
+        self.instrument_upsert_tests(batch_upsert_response, batch_upsert_request)
+
+        '''
+        Every instrument that is created is in LUSID given a unique LUSID Instrument Id or LUID for short. This ID is 
+        used for many methods and is how the API identifies an instrument. 
+
+        Note that there is also a match feature that allows you to search for instruments using the identifiers that we 
+        defined for each instrument, however it is preferred to use the LUID as this is guaranteed to be unique and 
+        there is absolutely no chance of a collision. 
+
+        We there want to add our newly created LUIDs to our initial holdings for future use
+        '''
+
+        # Loop over our recently upserted instruments
+        for result, instrument in batch_upsert_response.values.items():
```

Comment:
As discussed, create a dictionary from ClientInternal > LUID and then apply a map over transferred_instrument

Reply:
Realised that I can use the instrument name as the key and do it in one loop

`
        for instrument_name, instrument in batch_upsert_response.values.items():
            # Add our LUID as a new identifier so that we can use it in our calls later
            transferred_instrument[instrument_name]['identifiers']['LUID'] = instrument.lusid_instrument_id
`